### PR TITLE
i18n(es): Add translations for EC

### PR DESCRIPTION
**Pull request by kevinzunigacuellar**
#### Description

- Adds spanish translations for Expressive Code

### CODE REVIEW

File: .changeset/old-adults-bake.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/starlight': patch
+---
+
+Add spanish translations for Expressive Code UI
```
Comment (delucis) [inline .changeset/old-adults-bake.md:5]: Nit: uses Sarah’s recommended changeset verb tense and capitalises “Spanish”:

```suggestion
Adds Spanish translations for Expressive Code UI
```

**Review by delucis — APPROVED** [review]
Thanks @kevinzunigacuellar — LGTM!

cc @hippotastic — is the intention for us to collect these here or would PRs for these preferably happen directly in the `expressive-code` repo so all Expressive Code users can benefit?

**Comment by hippotastic** [comment]
> cc @hippotastic — is the intention for us to collect these here or would PRs for these preferably happen directly in the `expressive-code` repo so all Expressive Code users can benefit?

It's fine to let all translations happen here! I will wait for a couple of translations to accumulate and then take care of copying them over to EC.

File: packages/starlight/translations/es.json [hunk 2]
```diff
@@ -22,5 +22,8 @@
   "aside.note": "Nota",
   "aside.tip": "Consejo",
   "aside.caution": "Precaución",
-  "aside.danger": "Peligro"
+  "aside.danger": "Peligro",
+  "expressiveCode.copyButtonCopied": "Copiado",
```
Comment (hippotastic) [inline packages/starlight/translations/es.json:26]: Nit: The original text ends with an exclamation mark ("Copied!"), which I added to make it sound a little more affirmative/positive. I'd love this to be present in the translations as well!

Comment (delucis) [reply packages/starlight/translations/es.json:26]: Something like this @kevinzunigacuellar?

```suggestion
  "expressiveCode.copyButtonCopied": "¡Copiado!",
```

Comment (kevinzunigacuellar) [reply packages/starlight/translations/es.json:26]: ¡Sí, señor Chris!

**Review by delucis — APPROVED** [review]
¡Me parece bien! Gracias @kevinzunigacuellar 🙌

**Pull request merged**